music21/stream/tests.py

```diff
@@ -4726,6 +4738,13 @@ def testSortAndAutoSort(self):
         self.assertEqual([(0.0, 2), (0.0, 30), (5.0, 25), (8.0, 10), (10.0, 2),
                           (15.0, 10), (20.0, 2), (22.0, 1.0)], match)
 
+    def testInsertIgnoreSort(self):
+        '''A sorted stream does not become unsorted when ignoreSort=True.'''
+        s = Stream()
+        s.repeatAppend(note.Note(), 4)
+        s.insert(1, tempo.MetronomeMark(), ignoreSort=True)
+        self.assertTrue(s.isSorted)
```

Comment:
this seems strange to me -- can you explain this?  The Stream is no longer sorted -- why is it reporting True for isSorted?  I don't think I like this.  `isSorted` should be something that should be able to be trusted unless private methods are being used.

Reply:
Ah, I think I just botched the setup of the test case. Inserting something at the end would make a more clear test case, I'll fix it.

What motivated this: the docs for insert say:
> If `ignoreSort` is True then the inserting does not
        change whether the Stream is sorted or not (much faster if you're
        going to be inserting dozens
        of items that don't change the sort status)

But this test fails on master, which is to say, providing ignoreSort=True _does_ change the sort status, even if the stream stays sorted (as I'll update the test to actually demonstrate!) This is because insert() was unconditionally calling coreElementsChanged with clearIsSorted=True.

> isSorted should be something that should be able to be trusted unless private methods are being used.

Yeah, maybe ignoreSort needs to become exposed on coreInsert only?

Reply:
How I ended up here was:

1. I noticed that MIDI files have a tendency to produce simultaneities that don't collapse into chords, because of differing durations. So iterating elements and checking the element at i + 1 might still give you the same offset; I need the next distinct offset. I didn't want to have to look backwards in the elements array, so I realized I wanted sorted streams.
2. So I made MIDI parsing produce a stream with isSorted reported as True.
3. Then I went to add a test case to testQuantize, which builds streams with insert(). I then had to debug why the stream was reporting itself as unsorted. I chose between using ignoreSort=True versus doing an explicit sort.
4. ignoreSort=True wasn't preserving the value of isSorted (I agree that feels like a should-be-core thing, though)

Reply:
> Yeah, maybe ignoreSort needs to become exposed on coreInsert only?

yeah, I think that's a much better place.  And note that coreInsert still computes and returns whether the stream is guaranteed to be sorted.

But we shouldn't expose anything to the users where they could end up with a stream that needs manual sorting unless they set isSorted = False.  Feel free to remove.  Or just ignore the test and remove calls from this PR and I'll remove ignoreSort in insert later.

Reply:
Reverted changes and just went with an explicit sort 👍 . I think I misdiagnosed exactly what was unsorted in the first place anyway.